### PR TITLE
fix(Core:Battlegrounds): fixes some Arathi Basin achievements.

### LINKED ISSUE #7775: [Arathi Basin - Achievement] Territorial Dominance not granding part of achi when other faction
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [X] 40-49
- [ ] 50-59

### Current Behaviour

Original post: https://github.com/chromiecraft/chromiecraft/issues/1656

When you get in BG as other faction, not the one you play, you won't get credit for some achievements.
WoWScrnShot_090721_214046
WoWScrnShot_090721_214053

You can see on screenshots that I won BG with 5 bases on, and I did not get credit. This is like 5th time, would have had achievement done already 🗡️ 

### Expected Blizzlike Behaviour

Not possible on Blizzlike, as people are always their own faction when joining PvP BGs.

### Source

_No response_

### Steps to reproduce the problem

Enter BG as other faction by luck
Get 5 bases and win BG, get no credit.

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/595bb6adccbabc714469f3935541978283b8bdfb

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-corldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #7775

## Issues Addressed:

- Closes #7775
- Closes https://github.com/azerothcore/mod-cfbg/issues/48
- Closes https://github.com/chromiecraft/chromiecraft/issues/1658
- Closes https://github.com/chromiecraft/chromiecraft/issues/305

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. Enable cfbg module.
2. Enter Arathi Basin as other faction.
3. Get 5 bases and win BG, get no credit.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 1]
```diff
@@ -231,11 +231,11 @@ void BattlegroundMgr::BuildPvpLogDataPacket(WorldPacket* data, Battleground* bg)
...
 
     if (bg->GetStatus() != STATUS_WAIT_LEAVE)
-        *data << uint8(0);                                     // bg not ended
+        *data << uint8(0);                                          // bg not ended
     else
     {
-        *data << uint8(1);                                     // bg ended
-        *data << uint8(bg->GetWinner());                       // who win
+        *data << uint8(1);                                          // bg ended
+        *data << uint8(bg->GetWinner() == TEAM_ALLIANCE ? 1 : 0);   // who win
```
Comment (Helias) [inline src/server/game/Battlegrounds/BattlegroundMgr.cpp:238]: replace 1 and 0 with the TEAM_HORDE and TEAM_ALLIANCE

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 2]
```diff
@@ -235,7 +235,7 @@ void BattlegroundMgr::BuildPvpLogDataPacket(WorldPacket* data, Battleground* bg)
     else
     {
         *data << uint8(1);                                          // bg ended
-        *data << uint8(bg->GetWinner() == TEAM_ALLIANCE ? 1 : 0);   // who win
+        *data << uint8(bg->GetWinner() == TEAM_ALLIANCE ? TEAM_HORDE : TEAM_ALLIANCE); // who win
```
Comment (Winfidonarleyan) [inline src/server/game/Battlegrounds/BattlegroundMgr.cpp:238]: Maybe typo?

Comment (UltraNix) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp:238]: No? In packet it's reversed.

**Comment by Winfidonarleyan** [comment]
? Okay

**Review by Helias — APPROVED** [review]

**Comment by IntelligentQuantum** [comment]
Tested

**Review by IntelligentQuantum — APPROVED** [review]

**Pull request merged**